library/data/local/weaviate.py

```diff
@@ -191,6 +192,35 @@ def search(self, query:str, key: WeaviateSchemas, limit: int = 5, certainty: flo
 
         return response
     
+    def search_reranking(self, query:str, key: WeaviateSchemas, limit: int = 5, certainty: float = .7, threshold: float = .5) -> list[dict[str, any]]:
+        print("doing a reranking")
+        collection: Collection[Properties, References] = self.collection(key)
+
+        response: list[dict[str, any]] = collection.query.near_text(
+            query=query,
+            limit=limit,
+            certainty=certainty,
+            rerank=Rerank(
+                prop="text",
+                query=query
+            ),
+            return_metadata=wvc.query.MetadataQuery(distance=True)
+        ).objects
+
+        filtered_response = []
+        for o in response:
+            props = o.properties
+            rerank_score = o.metadata.rerank_score
+            normalized_rerank_score = 1/(1 + math.exp (-rerank_score))
+            props['normalized_rerank_score'] = normalized_rerank_score
```

Comment:
I'm not sure why I added this to props. I can remove it.

Reply:
The sigmoid was so we could threshold, not a bad idea to have the clarity on why a thing made it or didn't